aodntools/timeseries_products/aggregated_timeseries.py

```diff
@@ -121,6 +121,31 @@ def get_nominal_depth(nc):
 
     return nominal_depth
 
+def get_contributors(files_to_agg):
+    """
+    get the author and principal investigator details for each file
+
+    :param files_to_aggregate: list of files
+    :return: list: contributor_name, email and role
+    """
+
+    contributors = []
+    contributor_name, contributor_email, contributor_role = [], [], []
+
+    for file in files_to_agg:
+        with xr.open_dataset(file) as nc:
+            attributes = nc.attrs.keys()
+            if all(att in attributes for att in ['author', 'author_email']):
+                contributors.append((nc.author, nc.author_email, 'author'))
+            if all(att in attributes for att in ['principal_investigator', 'principal_investigator_email']):
+                contributors.append((nc.principal_investigator, nc.principal_investigator_email, 'principal_investigator'))
+
+    for item in set(contributors):
```

Comment:
If you actually want `contributors` as a set, why not make it one from the start? Just replace `contributors.append()` with `contributors.add()`.